src/utils/gauss_quad.hpp

```diff
@@ -0,0 +1,619 @@
+//  Copyright John Maddock 2015.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef GAUSS_QUAD_HPP
+#define GAUSS_QUAD_HPP
+
+#include <array>
+#include <cmath>
+#include <limits>
+
+#include "../kokkos_abstraction.hpp"
+
+namespace parthenon::math::quadrature {
+namespace detail {
+
+template <class T>
+struct gauss_constant_category {
+  static const unsigned value =
```

Comment:
is there a reason not to make this constexpr?

Reply:
fixed